python/lsst/eo/pipe/biasShiftsTask.py

```diff
@@ -0,0 +1,286 @@
+    return np.asarray(shift_peaks),  local_noise
+
+def satisfies_flatness(shiftrow, shiftmag, oscans, window=30, verbose=False):
+    
+    prerange = np.arange(shiftrow-window, shiftrow)
+    postrange = np.arange(shiftrow, shiftrow+window)
+    
+    preslope, preintercept, pre_r_value, p_value, std_err = scipy.stats.linregress(prerange,oscans[prerange])
+    postslope, postintercept, post_r_value, p_value, std_err = scipy.stats.linregress(postrange,oscans[postrange])
+    
+    if verbose:
+        print(f'Pre-threshold: {preslope*2*len(prerange)}')
+        print(f'Post-threshold: {postslope*2*len(postrange)}')
+        print(f'Shift value: {shiftmag}')
+    
+    pretrending  = (preslope*2*len(prerange) < shiftmag)   if (shiftmag>0) else (preslope*2*len(prerange) > shiftmag)
+    posttrending = (postslope*2*len(postrange) < shiftmag) if (shiftmag>0) else (postslope*2*len(postrange) > shiftmag)
+    
+    return (pretrending and posttrending)
+
+def find_shifts_in_exposure(exp, amp=None, method="conv", **options):
+    methods = ["derivative","conv"]
+    if method not in methods:
+        raise Exception(f'Shift detection method "{method}" not understood. Available methods are {methods}')        
+    if amp is None:
+        det = exp.getDetector()
+    else: 
+        det = [amp.getName()]
+    
+    shifts = {}
+
+    for thisAmp in det:
+        oscan_rows = get_oscan_rows(exp, thisAmp)
+        if method=="derivative":
+            shifts[thisAmp.getName()] = shift_in_rows(oscan_rows, **options)
+        if method=="conv":
+            shifts[thisAmp.getName()] = scan_for_shifts(oscan_rows, **options), np.mean(oscan_rows), np.mean(exp[thisAmp.getRawDataBBox()].image.array)
+    
+    if amp is None:
+        return shifts
+    else: 
+        return shifts[amp.getName()]
+    
+def butter_highpass(cutoff, fs, order=5):
+    nyq = 0.5 * fs
+    normal_cutoff = cutoff / nyq
+    b, a = scipy.signal.butter(order, normal_cutoff, btype='high', analog=False)
+    return b, a
+
+def butter_highpass_filter(data, cutoff, fs, order=5):
+    b, a = butter_highpass(cutoff, fs, order=order)
+    y = scipy.signal.filtfilt(b, a, data)
+    return y
+
+class BiasShiftsTaskConnections(pipeBase.PipelineTaskConnections,
+                            dimensions=("instrument", "detector")):
+    raws = cT.Input(name="raw",
+                    doc="Raw input frames",
+                    storageClass="Exposure",
+                    dimensions=("instrument", "exposure", "detector"),
+                    multiple=True,
+                    deferLoad=True)
+
+    bias_shifts = cT.Output(name="bias_shifts",
+                         doc="Sudden shifts in overscan biases",
+                         storageClass="DataFrame",
+                         dimensions=("instrument", "detector"))
+                         
+
+
+class BiasShiftsTaskConfig(pipeBase.PipelineTaskConfig,
+                        pipelineConnections=BiasShiftsTaskConnections):
+    pass
```

Comment:
The various functions defined in this file have a number of keyword options.  You might consider making the useful ones be configurable through this class.

Reply:
I'd like to add this functionality after getting a basic form into production.